cmake/FindUmpire.cmake

```diff
@@ -0,0 +1,25 @@
+# Find UMPIRE linear solver library
+#
+# Set UMPIRE_DIR to the base directory where the package is installed
+#
+# Sets two variables
+#   - UMPIRE_INCLUDE_DIRS
+#   - UMPIRE_LIBRARIES
+#
+
+find_path(UMPIRE_INCLUDE_DIRS
```

Comment:
Umpire uses CMake and has an install rule that exports a CMake file.  You shouldn't need this file, as CMake should find it automatically find it if you set `Umpire_ROOT`

Reply:
Ok. I see it in:
$UMPIRE_ROOT_DIR/share/umpire/cmake/umpire-config.cmake

However, cmake is not picking it up.

CMake Error at CMakeLists.txt:30 (find_package):
  By not providing "FindUmpire.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Umpire", but
  CMake did not find one.

  Could not find a package configuration file provided by "Umpire" with any
  of the following names:

    UmpireConfig.cmake
    umpire-config.cmake

  Add the installation prefix of "Umpire" to CMAKE_PREFIX_PATH or set
  "Umpire_DIR" to a directory containing one of the above files.  If "Umpire"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!

Reply:
Setting Umpire_DIR seems to work.

Reply:
Definitely remove the `FindUmpire.cmake` from the repos. It is preferable to use `umpire-config.cmake` as it is likely to provide modern CMake targets which have information regarding `INTERFACE_INCLUDE_DIRECTORIES` and `INTERFACE_LINK_LIBRARIES` which allow you to just do `target_link_libraries(hypre_app Umpire::umpire)` without having to do the `include_directories` step. It will also _forward_ any C++ flags it needs to downstream targets. 

It is better to explicitly set `-DUmpire_DIR=$UMPIRE_ROOT_DIR` in the `cmake` command (within exawind-builder script) so that it is explicit and also captured in `cmake-output.log` (useful for debugging). Don't do `export Umpire_DIR` and `unset`

Reply:
Setting -DUmpire_DIR in the cmake command is not sufficient. I changed the prefix path as shown in the most recent commit.